src/Domain/Services/OrderSourceAttribution.php

```diff
@@ -0,0 +1,199 @@
+	 * @param SourceAttributionController $source_attribution_controller Instance of the source attribution controller.
+	 */
+	public function __construct(
+		AssetApi $asset_api,
+		ExtendSchema $extend_schema,
+		FeaturesController $features_controller,
+		SourceAttributionController $source_attribution_controller
+	) {
+		$this->asset_api                     = $asset_api;
+		$this->features_controller           = $features_controller;
+		$this->extend_schema                 = $extend_schema;
+		$this->source_attribution_controller = $source_attribution_controller;
+	}
+
+	/**
+	 * Hook into WP.
+	 *
+	 * @since x.x.x
+	 * @return void
+	 */
+	public function init() {
+		if ( ! $this->features_controller->feature_is_enabled( 'order_source_attribution' ) ) {
+			return;
+		}
+
+		$this->extend_api();
+
+		// Bail early on admin requests to avoid asset registration.
+		if ( is_admin() ) {
+			return;
+		}
+
+		add_action(
+			'init',
+			function() {
+				$this->register_assets();
+			}
+		);
+
+		add_action(
+			'wp_enqueue_scripts',
+			function() {
+				$this->enqueue_scripts();
+			}
+		);
+	}
+
+	/**
+	 * Register scripts.
+	 */
+	private function register_assets() {
+		$this->asset_api->register_script(
+			'wc-blocks-order-source-attribution',
+			'build/wc-blocks-order-source-attribution.js',
+			[ 'woocommerce-order-source-attribution-js' ]
+		);
+	}
+
+	/**
+	 * Enqueue the Order Source Attribution script.
+	 *
+	 * @since x.x.x
+	 * @return void
+	 */
+	private function enqueue_scripts() {
+		wp_enqueue_script( 'wc-blocks-order-source-attribution' );
+	}
+
+	/**
+	 * Extend the Store API.
+	 *
+	 * @return void
+	 */
+	private function extend_api() {
+		try {
+			$this->extend_schema->register_endpoint_data(
+				[
+					'endpoint'        => CheckoutSchema::IDENTIFIER,
+					'namespace'       => 'woocommerce/order-source-attribution',
+					'schema_callback' => $this->get_schema_callback(),
+				]
+			);
+			$this->extend_schema->register_update_callback(
+				[
+					'namespace' => 'woocommerce/order-source-attribution',
+					'callback'  => function( $data ) {
+						// todo: Save the data to the order here.
+					},
+				]
+			);
+		} catch ( Exception $e ) { // phpcs:ignore Generic.CodeAnalysis.EmptyStatement
+			// Do nothing.
+		}
+	}
+
+	/**
+	 * Get the schema callback.
+	 *
+	 * @return callable
+	 */
+	private function get_schema_callback() {
+		return function() {
+			$schema = [];
+			$fields = $this->source_attribution_controller->get_fields();
+
+			$validate_callback = function( $value ) {
+				if ( ! is_string( $value ) && null !== $value ) {
+					return new WP_Error(
+						'api-error',
+						sprintf(
+							/* translators: %s is the property type */
+							esc_html__( 'Value of type %s was posted to the source attribution callback', 'woo-gutenberg-products-block' ),
+							gettype( $value )
+						)
+					);
+				}
+
+				return true;
+			};
+
+			$sanitize_callback = function( $value ) {
```

Comment:
If this is unlikely to change you can just pass `sanitize_text_field` as the `sanitize_callback` instead of using an inline function.